.github/workflows/run-build.yaml

```diff
@@ -0,0 +1,254 @@
+        required: false
+
+env:
+  BUILD_DIR: build-dir
+  ENVDIR: env.d
+
+jobs:
+  run-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout nextstrain/.github (ref ${{ inputs.ref }})
+        uses: actions/checkout@v3
+        with:
+          repository: nextstrain/.github
+          ref: ${{ inputs.ref }}
+
+      - name: Setup runtime ${{ inputs.runtime }}
+        uses: ./actions/setup-nextstrain-cli
+        with:
+          runtime: ${{ inputs.runtime }}
+        # Required to set up aws-batch as default runtime
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+      # envdir is not available in our Conda runtime
+      # (See https://github.com/nextstrain/conda-base/blob/135ec65cc286b6fc22f717a1157667cb9a7e0a55/src/recipe.yaml#L52)
+      # so it needs to be installed separately in the runtime environment.
+      #
+      # Installing directly in the nextstrain shell since pip installed packages
+      # do persist in the Conda runtime (https://github.com/nextstrain/cli/issues/282)
+      #   -Jover, 19 May 2023
+      - if: ${{ inputs.runtime == 'conda' }}
+        name: Install envdir for Conda runtime
+        shell: nextstrain shell . {0}
+        run: pip install envdir
+
+      - name: Checkout build repository
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.build-repo }}
+          path: ${{ env.BUILD_DIR }}
+
+      - if: inputs.env
+        name: Set environment variables
+        env:
+          env: ${{ inputs.env }}
+        run: >
+          # shellcheck disable=SC2154
+
+          echo "$env"
+          | ./bin/yaml-to-envvars
+          | tee -a "$GITHUB_ENV"
+
+      - if: inputs.runtime-envvar-names
+        name: Set secret runtime environment variables
+        env:
+          secrets: ${{ toJson(secrets) }}
+          varnames: ${{ inputs.runtime-envvar-names }}
+        run: >
+          # shellcheck disable=SC2154
+
+          echo "$secrets"
+          | ./bin/json-to-envvars --varnames="$varnames"
+          | tee -a "$GITHUB_ENV"
+
+      - name: Set build environment variables
+        env:
+          varnames: ${{ inputs.runtime-envvar-names }}
+        run: |
+          # shellcheck disable=SC2154
+
+          # read exits with 1 when it encounters EOF
+          read -a varnames -r -d $'\0' <<<"$varnames" || true
+          ./bin/write-envdir "${{ env.BUILD_DIR }}/${{ env.ENVDIR }}" "${varnames[@]}"
+
+      - name: Run build via ${{ inputs.runtime }}
+        env:
+          memory: ${{ inputs.memory && format('--memory "{0}"', inputs.memory) || '' }}
+          detach: ${{ inputs.runtime == 'aws-batch' && '--detach' || '' }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          nextstrain build \
+            --cpus ${{ inputs.cpus }} \
```

Comment:
When I tried to run a [workflow with a null value](https://github.com/joverlee521/nextstrain-testing/actions/runs/5192864216), it raised the following error message:
```
The template is not valid. .github/workflows/use-reusable.yaml (Line: 23, Col: 13): Unexpected value ''
```

This seems like the value is failing the type check since the cpus input is expected to be a number.